File: project/BBWT.Client/src/app/main/system-configuration/classes/pwa-settings.ts
import { SettingsSectionsName } from "../settings-sections-name";
import { SystemConfigurationResolveData } from "../system-configuration-resolve-data";

export class PwaSettings {
    desktopInstallationEnabled: boolean;
    mobileInstallationEnabled: boolean;
    desktopShowIndicator: boolean;
    mobileShowIndicator: boolean;

    static parse(data: SystemConfigurationResolveData): PwaSettings {
        return data.getConfigSection<PwaSettings>(SettingsSectionsName.PwaSettings);
    }
}

File: modules/BBWM.SystemSettings/PwaSettings.cs
namespace BBWM.SystemSettings;

public class PwaSettings : IMutableSystemConfigurationSettings
{
    public bool? DesktopInstallationEnabled { get; set; } = true;

    public bool? MobileInstallationEnabled { get; set; } = true;

    public bool? DesktopShowIndicator { get; set; } = true;

    public bool? MobileShowIndicator { get; set; } = true;
}
